### PR TITLE
[Bug] [Fix-10673] Workflow recovery bug with failed tasks

### PR DESCRIPTION
**Pull request by brave-lee**
closed by #10673

### CODE REVIEW

**Review by SbloodyS — APPROVED** [review]
LGTM

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java [hunk 1]
```diff
@@ -1125,6 +1125,22 @@ private boolean hasFailedTask() {
             if (errorTaskList.size() > 0) {
                 return true;
             }
+        } else {
+            if (processInstance.getCommandType() == CommandType.RECOVER_TOLERANCE_FAULT_PROCESS
+                || processInstance.getCommandType() == CommandType.RECOVER_SUSPENDED_PROCESS) {
+                List<Integer> failedList = processService.findTaskIdByInstanceState(processInstance.getId(), ExecutionStatus.FAILURE);
+                if (!failedList.isEmpty()) {
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1132]:
```suggestion
                if (CollectionUtil.isNotEmpty(failedList)) {
```

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java [hunk 2]
```diff
@@ -1125,6 +1125,22 @@ private boolean hasFailedTask() {
...
+                    return true;
+                }
+                List<Integer> toleranceList = processService.findTaskIdByInstanceState(processInstance.getId(), ExecutionStatus.NEED_FAULT_TOLERANCE);
+                if (!toleranceList.isEmpty()) {
+                    return true;
+                }
+                List<Integer> killedList = processService.findTaskIdByInstanceState(processInstance.getId(), ExecutionStatus.KILL);
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1139]: maybe kill != fail
image

Comment (brave-lee) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1139]: Please see ProcessService|constructProcessInstance() | line 798

Comment (brave-lee) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1139]: If the workflow contains killed, the workflow cannot be successful state

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java (hunk 1 above)
Comment (brave-lee) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1132]: I don't think it's necessary

Comment (caishunfeng) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java:1132]: Yes, some nip.

**Pull request merged**